### PR TITLE
QueryType string to enum

### LINKED ISSUE #1637: QuerySender queryType other or anything else not the same
**Describe the bug**
QuerySenders such as the FixedQuerySendere have an attribute `queryType`, which can be set to the following values according to the IbisDoc:

```
One of:
select for queries that return data
updateBlob for queries that update a BLOB
updateClob for queries that update a CLOB
package to execute Oracle PL/SQL package
other or anything else for queries that return no data.
```

It is implied that something that is not select/updateBlob/updateClob/package equals other, but this is not completely the case:
https://github.com/ibissource/iaf/blob/master/core/src/main/java/nl/nn/adapterframework/jdbc/JdbcQuerySenderBase.java#L161

The anything else also allows for a variety of non-existing queryTypes to be set.

**Reporter**
Ricardo

**Category** 
Use following Tags to categorize your bug report
 - Frank Framework

**To Reproduce**

**Expected behavior**
queryType anything to be truly the same as other, or have the possible queryTypes locked to select/updateBlob/updateClob/package/other and a warning/exception if queryType is not one from that list.

**Screenshots**
N/A

**Environment:**
 - Application server: [e.g. Tomcat/WebSphere/JBOSS/Jetty]: All
 - DBMS: [e.g. Oracle, MSSQL, MariaDB, MySql,PostgreSQL]: All
 - Browser: [e.g. Chrome, Safari, Edge]: All
 - IAF Version: [e.g. 7.5, 7.6]: All

**Additional context**
Add any other context about the problem here. (f.e. ladybug report / test adapter with larva test)

### PR DESCRIPTION
**Pull request by alisihab**
Closes #1637

### CODE REVIEW

File: core/src/main/java/nl/nn/adapterframework/jdbc/FixedQuerySender.java [hunk 1]
```diff
@@ -65,8 +66,8 @@ protected String getQuery(Message message) {
 
 	@Override
 	public boolean canProvideOutputStream() {
-		return  "updateClob".equalsIgnoreCase(getQueryType()) && StringUtils.isEmpty(getClobSessionKey()) ||
-				"updateBlob".equalsIgnoreCase(getQueryType()) && StringUtils.isEmpty(getBlobSessionKey());
+		return QueryType.UPDATECLOB==getQueryTypeEnum() && StringUtils.isEmpty(getClobSessionKey()) ||
+				QueryType.UPDATEBLOB==getQueryTypeEnum() && StringUtils.isEmpty(getBlobSessionKey());
```
Comment (gvanbrakel) [inline core/src/main/java/nl/nn/adapterframework/jdbc/FixedQuerySender.java:70]: When you compare strings to a fixed value, usually the fixed value is on the left side: ``"updateClob".equalsIgnoreCase(getQueryType())``
When testing enums, it is easier to read it the other way round:
 `` getQueryTypeEnum()==QueryType.UPDATECLOB``

File: core/src/main/java/nl/nn/adapterframework/jdbc/JdbcQuerySenderBase.java [hunk 2]
```diff
@@ -137,6 +137,14 @@
 	
 	private String convertedResultQuery;
 
+	public enum QueryType {
+		SELECT,
+		UPDATEBLOB,
+		UPDATECLOB,
+		PACKAGE,
+		OTHER,
```
Comment (gvanbrakel) [inline core/src/main/java/nl/nn/adapterframework/jdbc/JdbcQuerySenderBase.java:145]: Itn't this a comma too many?

**Review by gvanbrakel — CHANGES_REQUESTED** [review]

File: core/src/main/java/nl/nn/adapterframework/jdbc/JdbcQuerySenderBase.java [hunk 3]
```diff
@@ -316,39 +324,40 @@ protected PipeRunResult executeStatementSet(QueryExecutionContext queryExecution
...
+					return new PipeRunResult(null, executeUpdateBlobQuery(statement, message));
+				case UPDATECLOB:
+					if (StringUtils.isNotEmpty(getClobSessionKey())) {
+						return new PipeRunResult(null, executeUpdateClobQuery(statement,session==null?null:Message.asMessage(session.get(getClobSessionKey()))));
+					} 
+					return new PipeRunResult(null, executeUpdateClobQuery(statement, message));
+				case PACKAGE:
+					return new PipeRunResult(null, executePackageQuery(queryExecutionContext));
+				default:
+					break;
```
Comment (gvanbrakel) [inline core/src/main/java/nl/nn/adapterframework/jdbc/JdbcQuerySenderBase.java:358]: It is a good practice to 'do' something here, like:
- comment or print a log line that this case is expected and does not need special handling
- throw an IllegalStateException, if this case should not be possible. That helps to fix problems when OutputTypes change over the years

**Review by gvanbrakel — CHANGES_REQUESTED** [review]

**Review by gvanbrakel — APPROVED** [review]

File: core/src/test/java/nl/nn/adapterframework/util/LockerTest.java [hunk 4]
```diff
@@ -22,6 +22,7 @@
 import nl.nn.adapterframework.core.IbisTransaction;
 import nl.nn.adapterframework.jdbc.JdbcException;
 import nl.nn.adapterframework.jdbc.TransactionManagerTestBase;
+import nl.nn.adapterframework.jdbc.JdbcQuerySenderBase.QueryType;
```
Comment (nielsm5) [inline core/src/test/java/nl/nn/adapterframework/util/LockerTest.java:25]: Do you use this import anywhere?

File: core/src/main/java/nl/nn/adapterframework/jdbc/QueryExecutionContext.java [hunk 5]
```diff
@@ -45,18 +47,17 @@ public void setQuery(String query) {
 		this.query = query;
 	}
 
-	public String getQueryType() {
+	public QueryType getQueryTypeEnum() {
```
Comment (nielsm5) [inline core/src/main/java/nl/nn/adapterframework/jdbc/QueryExecutionContext.java:50]: This method doesn't require the Enum suffix?

File: core/src/main/java/nl/nn/adapterframework/jdbc/QueryExecutionContext.java [hunk 6]
```diff
@@ -45,18 +47,17 @@ public void setQuery(String query) {
...
 		return queryType;
 	}
 	public void setQueryType(String queryType) {
-		this.queryType = queryType;
+		this.queryType = Misc.parse(QueryType.class, "queryType", queryType);
 	}
```
Comment (nielsm5) [inline core/src/main/java/nl/nn/adapterframework/jdbc/QueryExecutionContext.java:55]: Is this method used anywhere? Isn't it easier to pass the Enum?

**Review by nielsm5 — APPROVED** [review]

**Pull request merged**